File: 3412-permutation-difference-between-two-strings/permutation-difference-between-two-strings.ts
function findPermutationDifference(s: string, t: string): number {
    let difference = 0;
    let sIndexMap: {[key: string]: number} = {};

    for (let i = 0; i < s.length; i++) {
        sIndexMap[s[i]] = i;
    }

    for (let i = 0; i < t.length; i++) {
        difference += Math.abs(sIndexMap[t[i]] - i);
    }

    return difference;
}

File: 3412-permutation-difference-between-two-strings/permutation-difference-between-two-strings.go
func findPermutationDifference(s string, t string) int {
    var diff int
    for i := 0; i < len(s); i++ {
        for j := 0; j < len(t); j++ {
            if s[i] == t[j] {
                diff += abs(i - j)
                break
            }
        }
    }
    return diff
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}